src/openrct2/ride/RideData.h

```diff
@@ -142,16 +142,46 @@ struct UpkeepCostsDescriptor
     uint8_t CostPerStation;
 };
 
+constexpr uint32_t trackGroupDigits = std::numeric_limits<uint32_t>::digits;
+constexpr uint32_t trackGroupMask = std::numeric_limits<uint32_t>::digits - 1;
+
+struct ride_track_group_t
+{
+    uint32_t data[(TRACK_GROUP_COUNT + trackGroupMask) / trackGroupDigits] = {};
+    bool IsTrackEnabled(int32_t trackFlagIndex) const
+    {
+        return (data[trackFlagIndex / trackGroupDigits] & (1 << (trackFlagIndex & trackGroupMask))) != 0;
+    };
+    constexpr void SetTrackEnabled(int32_t trackFlagIndex, bool enabled)
+    {
+        data[trackFlagIndex / trackGroupDigits] &= ~(1 << (trackFlagIndex & trackGroupMask));
+        data[trackFlagIndex / trackGroupDigits] |= enabled * (1 << (trackFlagIndex & trackGroupMask));
+    };
+    constexpr void EnableTrack(int32_t trackFlagIndex)
+    {
+        data[trackFlagIndex / trackGroupDigits] |= (1 << (trackFlagIndex & trackGroupMask));
+    };
+    constexpr void DisableTrack(int32_t trackFlagIndex)
+    {
+        data[trackFlagIndex / trackGroupDigits] &= ~(1 << (trackFlagIndex & trackGroupMask));
+    };
+    constexpr ride_track_group_t(std::initializer_list<uint16_t> il)
+    {
+        for (uint16_t i : il)
+            EnableTrack(i);
+    };
+};
+
```

Comment:
I was thinking a bit about how this simplify this, you can basically remove the entire wrapper.
I think a more elegant solution would be doing
```
using RideTrackGroup = std::bitset<64>;
```
And as for the SetField problem
```
constexpr RideTrackGroup MakeTrackGroup(const std::initializer_list<uint16_t>& vals)
{
 // Construct from vals...
}
...
SET_FIELD(EnabledTrackPieces, MakeTrackGroup({})),
```
By doing this we can continue to use all bit operations such as  or, xor, and etc.

Reply:
Duncan and I discussed bitset in the discussion regarding this pull request.

Reply:
My concern with bitset was serialisation and initialising. @ZehMatt is suggesting a solution to the initialisation issue and the serialisation part of it is not an issue (for now).

Reply:
Is calling the function in the SET_FIELD function and using the return value substantially different from https://github.com/OpenRCT2/OpenRCT2/pull/16172/commits/422f55796b50475d912abe0bc2c8ac2009a62d21 with regards to intellisense whining about using non-constexpr method 'set'?

Reply:
Apparently std::bitset is pretty shit, I looked up the documentation and only a single constructor has the constexpr keyword. We should probably build our own bitset in this case, fortunately that is rather easy to do.